site-packages/integralstor/system_info.py

```diff
@@ -51,4 +52,221 @@ def load_system_config(first_time=False):
+        is_del = True
+
+    except Exception, e:
+        return None, 'Could not delete UUID: %s' % e
+    else:
+        return is_del, None
+
+
+def set_integralstor_uuid(regenerate=False):
+    """Set Integralstor's UUID fields
+
+    args:       Bool:regenerate
+    returns:    dict with keys 'uuid_hex' & 'uuid_str'
+
+    """
+    ret_id = None
+    try:
+        uuid = None
+        db_path, err = config.get_db_path()
+        if err:
+            raise Exception(err)
+        uuid, err = get_integralstor_uuid()
+        if err:
+            raise Exception(err)
+        if uuid and regenerate is False:
+            # UUID is already set, do not regenerate
+            ret_id = uuid
+
+        if not uuid or regenerate is True:
+            # Generate UUID
+            uuid = UUID.uuid1()
+
+            # Clear the existing entry from the table
+            is_del, err = delete_integralstor_uuid()
+            if err:
+                raise Exception('Could not clear the existing entry: %s' % err)
+
+            cmd = "insert into integralstor_info (uuid_hex,uuid_str) values ('%s','%s')" % (
+                str(uuid.hex), str(uuid))
+            row_id, err = db.execute_iud(
+                db_path, [[cmd], ], get_rowid=True)
+            if err:
+                raise Exception(err)
+            ret_id = {}
+            ret_id['uuid_hex'] = str(uuid.hex)
+            ret_id['uuid_str'] = str(uuid)
+
+    except Exception, e:
+        return None, 'Could not set UUID: %s' % e
+    else:
+        return ret_id, None
+
+
+def get_integralstor_uuid():
+    """Return Integralstor's UUID fields
+
+    args:       None
+    returns:    dict with keys 'uuid_hex' & 'uuid_str'
+
+    """
+    id_dict = None
+    try:
+        db_path, err = config.get_db_path()
+        if err:
+            raise Exception(err)
+
+        cmd = "select * from integralstor_info"
+        ret, err = db.get_single_row(db_path, cmd)
+        if err:
+            raise Exception(err)
+        if ret:
+            # return the dict which has all the fields
+            id_dict = ret
+    except Exception, e:
+        return None, 'Could not fetch UUID: %s' % e
+    else:
+        return id_dict, None
+
+
+def get_org_info():
+    """Return organization's information
+
+    args:       None
+    returns:    dict with keys-
+                    'org_name',
+                    'unit_name',
+                    'unit_id',
+                    'subunit_name',
+                    'subunit_id'
+    """
+    ret_dict = {}
+    ret_dict['org_name'] = 'N.A'
+    ret_dict['unit_name'] = 'N.A'
+    ret_dict['unit_id'] = 'N.A'
+    ret_dict['subunit_name'] = 'N.A'
+    ret_dict['subunit_id'] = 'N.A'
+    try:
+        db_path, err = config.get_db_path()
+        if err:
+            raise Exception(err)
+
+        cmd = "select * from org_info"
+        ret, err = db.get_single_row(db_path, cmd)
+        if err:
+            raise Exception(err)
+        if ret:
```

Comment:
It might be better to return a None if there is no org entry in the DB. The reason is that then it is easier in the template to say that it has not been set. Also for the reports, we can detect that it has not been set more easily and decide whether to pull in that info into a report.

Reply:
Okay, updated.